src/vmm/tests/integration_tests.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate devices;
+extern crate polly;
+extern crate vm_memory;
+extern crate vmm;
+extern crate vmm_sys_util;
+
+mod mock_devices;
+
+use std::io;
+
+use polly::event_manager::EventManager;
+use vmm::builder::setup_serial_device;
+use vmm_sys_util::tempfile::TempFile;
+
+use mock_devices::MockSerialInput;
```

Comment:
nit:

I see multiple modules `mock_devices/mod.rs`, `mock_resources/mod.rs`.

How about a single module `mock_objects`, which has different inner files/modules like `devices.rs`, `resources-configs.rs`, etc? I think it'd be somewhat easier to navigate.

disclaimer: this is not a strong opinion, just an idea.

Reply:
That's how I wanted to organize them initially, but `cargo` considers that everything not in `mod.rs` is a test and runs it as such. Not a big deal necessarily, but any functions there would confusingly show up in test output as passed (hopefully) tests. See [details](https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests).